### DISCUSSION TITLE
KKfileview预览服务接入，无需部署中间件及修改页面元素

### DISCUSSION THREAD

**Discussion by zanzhz1101** [AlistGo/alist#6970] [General]
首先部署好kkfileview，官方文档https://kkfileview.keking.cn/zh-cn/docs/home.html
我的本地部署服务器为192.168.0.21，预览服务接入地址为http://192.168.0.21:8012/onlinePreview?url='+encodeURIComponent(base64Encode(url))
Alist==》Alist管理==》设置==》预览==》Iframe 预览
删除原有预览，填入以下内容：

> {
    "doc": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "docx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xls": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xlsx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xlsm": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ppt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "pptx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "csv": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "tsv": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dotm": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xlt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xltm": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dot": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dotx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xlam": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xla": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "pages": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "wps": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dps": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "et": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ett": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "wpt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "odt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ods": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ots": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "odp": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "otp": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "six": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ott": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "fodt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "fods": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "vsd": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "vsdx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "psd": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "eps": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "pdf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ofd": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "rtf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "xmind": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "epub": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "obj": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "3ds": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "stl": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ply": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "gltf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "glb": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "off": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "3dm": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "fbx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dae": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "wrl": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "3mf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ifc": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "brep": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "step": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "iges": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "fcstd": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "bim": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dwg": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dxf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dwf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "igs": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dwt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dng": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "dwfx": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "cf2": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "plt": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "zip": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "rar": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "jar": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "tar": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "gzip": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "7z": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "tif": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "tiff": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "svg": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "avi": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "mov": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "rm": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "webm": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ts": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "mkv": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "mpeg": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "ogg": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "mpg": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "rmvb": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "wmv": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "3gp": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    },
    "swf": {
        "预览": "http://192.168.0.21:8012/onlinePreview?url=$eb_url"
    }
}

> 

保存即可实现文件预览
![image](https://github.com/user-attachments/assets/978c54e7-c58a-48c1-9a5f-c1d211d24107)

**Comment by snowdream** [comment]
非常棒的方案，感谢分享。

**Reply by forvenus** [reply]
感谢感谢,局域网nas不用折腾https+cors, pdf/office都能搞定,完美啊

**Comment by kk066** [comment]
当文件名是中文的时候会返回401报错呢

**Reply by leiddev** [reply]
这个问题咋解决？

**Reply by zanzhz1101** [reply]
尝试改成$eb_durl？如果能行的话麻烦发一下，我这里目前没有环境了

**Reply by leiddev** [reply]
试过$eb_durl了也不行

**Reply by zanzhz1101** [reply]
目测应该是kk回源的链接与实际不一致，看看实际是啥吧

**Reply by zanzhz1101** [reply]
基础变量：

$url: the file url
$durl: the file of alist URL
$name: the file name
拓展变量：
在基础变量前添加[eb_]，其中e表示encodeURIComponent，b表示base64， 如

$e_url: encodeURIComponent($url)
$b_url: btoa($url)
$eb_url: encodeURIComponent(btoa($url))

**Reply by leiddev** [reply]
使用nginx部署view.html的中转方案了，可以成功。

**Comment by BoyInTheSun** [comment]
`$eb_url`应改为`$eb_durl`，否则，当网盘直链不是以原后缀名结尾时，会导致不识别文件格式。参考[文档](https://alist.nn.ci/zh/config/preview.html#%E5%A4%96%E9%83%A8%E9%A2%84%E8%A7%88)。
